merge_objects.py

```diff
@@ -0,0 +1,51 @@
+import unittest
+
+
+def merge(obj_a, obj_b):
+    if obj_a is None or obj_b is None:
+        return obj_a or obj_b
+    if type(obj_a) != type(obj_b):
+        return obj_a, obj_b
+    if isinstance(obj_a, (set, frozenset)):
+        return obj_a | obj_b
+    if isinstance(obj_a, dict):
+        return merge_dict(obj_a, obj_b)
+    return obj_a + obj_b
+
+
+def merge_dict(obj_a, obj_b):
+    result = {}
+    for k in list(obj_a) + list(obj_b):
+        result[k] = merge(obj_a.get(k), obj_b.get(k))
+    return result
+
+
+class TestMerge(unittest.TestCase):
+    def setUp(self):
+        self.__obj_a = {'x': [1, 2, 3], 'y': 1, 'z': set([1, 2, 3]),
```

Comment:
We don't need to be that strict as in Java with the private fields. Therefore there are rare cases when we define private fields using one of the two conventions _ (private instance field) __(private class fields)